Using the coverage reporter requires "@hapi/eslint-plugin" to be installed, even if it’s not used.

#### Support plan

* *is this issue currently blocking your project?* (yes/no): no
* *is this issue affecting a production system?* (yes/no): no

#### Context

* *node version*: 14.17.1
* *module version with issue*: 24.3.1
* *last module version without issue*: at least 24.1.1, maybe newer
* *environment* (e.g. node, browser, native): node
* *used with* (e.g. hapi application, another framework, standalone, ...): hapi application, but it shouldn’t matter
* *any other relevant information*: -

#### What are you trying to achieve or the steps to reproduce?

Run lab with the coverage reporter and disabled linting in a project without `@hapi/eslint-plugin` installed:
```
./node_modules/.bin/lab -c
```

#### What was the result you got?

You get the following error:
```
UnhandledPromiseRejectionWarning: Error: Failed to load plugin '@hapi' declared in 'BaseConfig': Cannot find module '@hapi/eslint-plugin'
```

This wasn’t the case at least until 24.1.1.

#### What result did you expect?

Lab shouldn’t require the eslint plugin if linting isn’t enabled.

I've run into this same issue with @hapi/lab@24.3.2 & eslint@8.0.1, node 14.17.0 & npm 7.24.0. The error seems related to both this and https://github.com/hapijs/lab/issues/1021, because I don't get the same error if I downgrade eslint to 7.x.x

npm test command: `lab --verbose --colors -S -r console -r html -o stdout -o coverage.html -a @hapi/code -p 1`

relavent dependency tree:
```
➜ npm ls eslint @hapi/eslint-plugin @hapi/lab

myproject@2.0.0 /Users/myuser/Repos/myproject
├─┬ @myprivateorg/eslint-config@2.3.2
│ └── eslint@8.0.1 deduped
├─┬ @hapi/lab@24.3.2
│ ├─┬ @babel/eslint-parser@7.15.8
│ │ └── eslint@7.32.0 deduped
│ ├── @hapi/eslint-plugin@5.1.0
│ └── eslint@7.32.0
└─┬ eslint@8.0.1
  └─┬ eslint-utils@3.0.0
    └── eslint@8.0.1 deduped
```

And the error we get (which exits with code 0 so isn't getting flagged as a fail by our CI server 😕  )

```
(node:59134) UnhandledPromiseRejectionWarning: Error: Failed to load plugin '@hapi' declared in 'BaseConfig': Cannot find module '@hapi/eslint-plugin'
Require stack:
- /Users/myuser/Repos/myproject/__placeholder__.js
Referenced from: BaseConfig
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
    at Function.resolve (internal/modules/cjs/helpers.js:94:19)
    at Object.resolve (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:28:50)
    at ConfigArrayFactory._loadPlugin (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:1017:39)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:837:29)
    at ConfigArrayFactory._loadExtends (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:779:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:720:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory.create (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/config-array-factory.js:460:16)
    at createBaseConfigArray (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:98:48)
    at new CascadingConfigArrayFactory (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:234:30)
    at new CLIEngine (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/node_modules/eslint/lib/cli-engine/cli-engine.js:569:36)
    at Object.<anonymous> (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/lib/modules/coverage.js:21:19)
    at Module._compile (internal/modules/cjs/loader.js:1068:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:933:32)
    at Function.Module._load (internal/modules/cjs/loader.js:774:14)
    at Module.require (internal/modules/cjs/loader.js:957:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.get [as coverage] (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/lib/modules/index.js:15:20)
    at Object.exports.run (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/lib/cli.js:38:17)
    at main (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/bin/lab:56:48)
    at Object.<anonymous> (/Users/myuser/Repos/myproject/node_modules/@hapi/lab/bin/lab:60:1)
    at Module._compile (internal/modules/cjs/loader.js:1068:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:933:32)
    at Function.Module._load (internal/modules/cjs/loader.js:774:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
(Use `node --trace-warnings ...` to show where the warning was created)
(node:59134) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:59134) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

--- a/package.json
+++ b/package.json
@@ -36,7 +36,8 @@
         "will-call": "1.x.x"
     },
     "peerDependencies": {
-        "typescript": ">=3.6.5"
+        "typescript": ">=3.6.5",
+        "@hapi/eslint-plugin": "^5.1.0"
     },
     "peerDependenciesMeta": {
         "typescript": {

